chirp/drivers/icm710.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2024 Paul Decker <kg7hf1@gmail.com>
+#      Derives from template.py
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import struct
+import time
+import logging
+from collections import OrderedDict
+from tkinter import N
+
+from chirp import chirp_common, directory, memmap, errors, util
+from chirp import bitwise
+from chirp.drivers import icf
+from chirp.settings import RadioSettingGroup, RadioSetting
+from chirp.settings import RadioSettingValueBoolean, RadioSettingValueList
+from chirp.settings import RadioSettingValueString, RadioSettingValueInteger
+from chirp.settings import RadioSettings, RadioSettingSubGroup
+
+
+
+
+LOG = logging.getLogger(__name__)
+
+
+BAUDS = [4800]
+POWER_LEVELS = []
+#TODO: Need to find where power is saved in the image
+#POWER_LEVELS = [chirp_common.PowerLevel('Low', dBm=43),  #20 watts
+#                chirp_common.PowerLevel('Med', dBm=48),  #60 watts
+#                chirp_common.PowerLevel('High', dBm=52)] #150 watts
+DUPLEX_MODES = ['split', 'off']                
+MODULATION_MODES = ["USB", "R3E", "AM", "LSB", "AFS", "FSK", "CW"]  #do not reorder    
+TUNING_STEPS = [0.5]
+
+MEM_FORMAT = """
+
+#seekto 0xE020;
+struct {
+  bbcd rx_freqL[4];   
+  u8 modeL;
+  char nameL[7]; 
+  bbcd tx_freqL[4];  
+  bbcd rx_freqH[4];   
+  u8 modeH;
+  char nameH[7]; 
+  bbcd tx_freqH[4];    
+}memory[233];  
+
+"""
+
+
+
+
+    
+@directory.register
+class IcomM710Radio(icf.IcomCloneModeRadio):
+    """ICOM IC-M710"""
+    VENDOR = "Icom"
+    MODEL = "M710"
+    BAUD_RATE = 4800
+    
+    _model = "\x162\x00\x01"  # 4-byte model string
+    _endframe = ''
+    _ignore_clone_ok = True
+
+    _memsize = 0xEFC0 
+    _ranges = [(0xE000, 0xEFC0, 32)]                 # Ranges of the mmap to send to the radio
+
+    _num_banks = 10              # Most simple Icoms have 10 banks, A-J
+    _bank_index_bounds = (0, 99)
+ #   _bank_class = IcomBank
```

Comment:
I know this is a draft, but please do remove commented-out code from the version we expect to merge.

Reply:
Done